File: Components/DevExpressVCL/Demos/ExpressQuantumTreeList/Delphi/Common/DemoBasicMain.pas
unit DemoBasicMain;

{$I cxVer.inc}

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, cxControls, cxLookAndFeels, ActnList, ImgList, Menus, ComCtrls,
{$IFDEF EXPRESSBARS}
  dxBar, dxStatusBar,
{$ENDIF}
  StdCtrls, cxTL, cxTLExportLink, SkinDemoUtils;

type
  TDemoBasicMainForm = class(TForm)
    ilMain: TImageList;
    alMain: TActionList;
    actHelp: TAction;
    actProducts: TAction;
    actDownloads: TAction;
    actForum: TAction;
    actDXOnTheWeb: TAction;
    actAbout: TAction;
    actExit: TAction;
    sbMain: TStatusBar;
    lscrip: TLabel;
    actShowDemoDescription: TAction;
    cxLookAndFeelController: TcxLookAndFeelController;
    actRateDemo: TAction;
    actSupport: TAction;
    mmMain: TMainMenu;
    miFile: TMenuItem;
    miExit: TMenuItem;
    miOptions: TMenuItem;
    miSeparator2: TMenuItem;
    miShowDemoDescription: TMenuItem;
    miHelp: TMenuItem;
    miGridHelp: TMenuItem;
    miRate: TMenuItem;
    miSeparator3: TMenuItem;
    miDeveloperExpressontheweb: TMenuItem;
    miProducts: TMenuItem;
    miDownloads: TMenuItem;
    miSupport: TMenuItem;
    miForum: TMenuItem;
    miSeparator4: TMenuItem;
    miAbout: TMenuItem;
    Exportto1: TMenuItem;
    excel1: TMenuItem;
    html1: TMenuItem;
    text1: TMenuItem;
    xml1: TMenuItem;
    SaveDialog1: TSaveDialog;
    Excel2: TMenuItem;
    miTouchMode: TMenuItem;
    procedure actDXOnTheWebExecute(Sender: TObject);
    procedure actExitExecute(Sender: TObject);
    procedure actShowDemoDescriptionExecute(Sender: TObject);
    procedure actHelpExecute(Sender: TObject);
    procedure actRateDemoExecute(Sender: TObject);
    procedure actAboutExecute(Sender: TObject);
    procedure ExportToClick(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure miTouchModeClick(Sender: TObject);
  private
    function GetTreeList: TcxCustomTreeList;
  public
  {$IFDEF EXPRESSBARS}
    BarManager: TdxBarManager;
  {$ENDIF}
    procedure AfterConstruction; override;
  end;

var
  DemoBasicMainForm: TDemoBasicMainForm;

implementation

{$R *.dfm}

uses
  AboutDemoForm, DemoRating, DemoUtils;

procedure TDemoBasicMainForm.AfterConstruction;
begin
  inherited;
{$IFDEF EXPRESSBARS}
  BarManager := TdxBarManager.Create(Self);
  dxBarConvertMainMenu(mmMain, BarManager);
  BarManager.Style := bmsUseLookAndFeel;
  BarManager.MainMenuControl.DockControl.Top := 0;
{$ENDIF}
{$IFDEF EXPRESSBARS}
  CreateSkinsMenuItem(BarManager);
{$ELSE}
  CreateSkinsMenuItem(mmMain);
{$ENDIF}
  lscrip.Transparent := False;
end;

procedure TDemoBasicMainForm.actDXOnTheWebExecute(
  Sender: TObject);
begin
  ShowWebPage(TdxWebPageType(TComponent(Sender).Tag));
end;

procedure TDemoBasicMainForm.actExitExecute(Sender: TObject);
begin
  Close;
end;

procedure TDemoBasicMainForm.actShowDemoDescriptionExecute(
  Sender: TObject);
begin
  lscrip.Visible := not lscrip.Visible;
  TCustomAction(Sender).Checked := not TCustomAction(Sender).Checked;
end;

procedure TDemoBasicMainForm.actHelpExecute(Sender: TObject);
begin
  Application.HelpCommand(HELP_FINDER, 0);
end;

procedure TDemoBasicMainForm.actRateDemoExecute(Sender: TObject);
begin
  with TDemoRatingForm.Create(Self) do
    try
      ShowModal;
    finally
      Free;
    end;
end;

procedure TDemoBasicMainForm.actAboutExecute(Sender: TObject);
begin
  ShowAboutDemoForm;
end;

procedure TDemoBasicMainForm.ExportToClick(Sender: TObject);
var
  ATreeList: TcxCustomTreeList;
begin
  ATreeList := GetTreeList;
  if (ATreeList <> nil) and SaveDialog1.Execute then
    case TMenuItem(Sender).Tag of
      0:
        cxExportTLToExcel(SaveDialog1.FileName, ATreeList);
      1:
        cxExportTLToText(SaveDialog1.FileName, ATreeList);
      2:
        cxExportTLToHTML(SaveDialog1.FileName, ATreeList);
      3:
        cxExportTLToXML(SaveDialog1.FileName, ATreeList);
      4:
        cxExportTLToXLSX(SaveDialog1.FileName, ATreeList);
    end;
end;

function TDemoBasicMainForm.GetTreeList: TcxCustomTreeList;
var
  I: Integer;
begin
  Result := nil;
  for I := 0 to ControlCount - 1 do
    if Controls[I] is TcxCustomTreeList then
    begin
      Result := TcxCustomTreeList(Controls[I]);
      Break;
    end;
end;

procedure TDemoBasicMainForm.miTouchModeClick(Sender: TObject);
begin
  cxLookAndFeelController.TouchMode := GetMenuItemChecked(Sender);
end;

procedure TDemoBasicMainForm.FormCreate(Sender: TObject);
begin
  miTouchMode.Checked := cxIsTouchModeEnabled;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressQuantumTreeList/CBuilder/Common/DemoBasicMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "DemoBasicMain.h"
#include "DemoRating.h"
#include "AboutDemoForm.h"
#include "DemoUtils.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxControls"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma resource "*.dfm"
TDemoBasicMainForm *DemoBasicMainForm;
//---------------------------------------------------------------------------
__fastcall TDemoBasicMainForm::TDemoBasicMainForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actAboutExecute(TObject *Sender)
{
  ShowAboutDemoForm();
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::miTouchModeClick(TObject *Sender)
{
  cxLookAndFeelController->TouchMode = miTouchMode->Checked;
}

void __fastcall TDemoBasicMainForm::actDXOnTheWebExecute(TObject *Sender)
{
  ShowWebPage((TdxWebPageType)(((TComponent*)Sender)->Tag));
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actExitExecute(TObject *Sender)
{
  Close();
}
//---------------------------------------------------------------------------


void __fastcall TDemoBasicMainForm::actHelpExecute(TObject *Sender)
{
  Application->HelpCommand(HELP_FINDER, 0);
}
//---------------------------------------------------------------------------


void __fastcall TDemoBasicMainForm::actRateDemoExecute(TObject *Sender)
{
  TDemoRatingForm *AForm = new TDemoRatingForm(this);
    __try{
      AForm->ShowModal();
    }
    __finally{
      AForm->Free();
    }
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actShowDemoDescriptionExecute(TObject *Sender)
{
  lscrip->Visible = !lscrip->Visible;
  ((TCustomAction*)Sender)->Checked = !((TCustomAction*)Sender)->Checked;
}
//---------------------------------------------------------------------------



void __fastcall TDemoBasicMainForm::ExportToClick(TObject *Sender)
{
  TcxCustomTreeList* ATreeList = GetTreeList();
  if ((ATreeList != NULL) && SaveDialog1->Execute())
	switch (((TMenuItem*)Sender)->Tag){
	  case 0:
		cxExportTLToExcel(SaveDialog1->FileName, ATreeList); break;
	  case 1:
		cxExportTLToText(SaveDialog1->FileName, ATreeList); break;
	  case 2:
		cxExportTLToHTML(SaveDialog1->FileName, ATreeList); break;
	  case 3:
	   	cxExportTLToXML(SaveDialog1->FileName, ATreeList); break;
	  case 4:
	   	cxExportTLToXLSX(SaveDialog1->FileName, ATreeList); break;
	 };
}
//---------------------------------------------------------------------------

TcxCustomTreeList* __fastcall TDemoBasicMainForm::GetTreeList()
{
  for (int i = 0; ControlCount - 1; i++)
  {
	if (dynamic_cast<TcxCustomTreeList*>(Controls[i]) != NULL)
	  return (TcxCustomTreeList*)(Controls[i]);
  };
  return NULL;
}
void __fastcall TDemoBasicMainForm::FormCreate(TObject *Sender)
{
  SetDefaultLookAndFeel();
  AddLookAndFeelMenu();
  miTouchMode->Checked = cxIsTouchModeEnabled;
}

void __fastcall TDemoBasicMainForm::AddLookAndFeelMenu()
{
  miOptions->Insert(miOptions->IndexOf(miSeparator2),
	CreateLookAndFeelMenuItems(miOptions, cxLookAndFeelController));
}

TcxLookAndFeelKind __fastcall TDemoBasicMainForm::GetDefaultLookAndFeelKind()
{
  return(lfUltraFlat);
}

bool __fastcall TDemoBasicMainForm::IsNativeDefaultStyle()
{
  return(true);
}

void __fastcall TDemoBasicMainForm::SetDefaultLookAndFeel()
{
  cxLookAndFeelController->NativeStyle = IsNativeDefaultStyle();
  cxLookAndFeelController->Kind = GetDefaultLookAndFeelKind();
}

//---------------------------------------------------------------------------

